homeassistant/components/tplink/const.py

```diff
@@ -3,15 +3,36 @@
 
 from typing import Final
 
-from homeassistant.const import Platform
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_AUTHENTICATION,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    Platform,
+)
 
 DOMAIN = "tplink"
+STORAGE_KEY: Final = DOMAIN
+STORAGE_VERSION: Final = 1
+STORAGE_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_AUTHENTICATION): {
+            vol.Required(CONF_USERNAME): str,
+            vol.Required(CONF_PASSWORD): str,
+        }
+    }
+)
+
+DISCOVERY_TIMEOUT = 8  # Home Assistant will complain if startup takes > 10s
```

Comment:
```suggestion
DISCOVERY_TIMEOUT = 5  # Home Assistant will complain if startup takes > 10s
```

Ideally we leave this at 5 and let devices retry after startup since we don't want to block startup for too long

Reply:
See https://ptb.discord.com/channels/@me/1194692089187926067/1197723405949079602

Reply:
Updated to 5